### ISSUE TITLE
Cannot backspace-delete in web mode

### ISSUE THREAD

**Issue by skosch** [hlissner/doom-emacs#712]
When in web-mode, RJSX, or similar, I occasionally cannot delete characters using backspace.

For instance, create a new file, turn on web-mode, and enter (`|` is your cursor):
```jsx
(
<|>
)
<>
```
then hit backspace. Instead of deleting the `<`, I get an error `Args out of range: #("<" 0 1 (part-side jsx ...`.

If this is an issue with one of the mode packages, I'll close this issue and ask there.

Thanks :)

### System information

<details>
<summary>Click to expand</summary>
```
- OS: gnu/linux (x86_64-pc-linux-gnu)
- Emacs: 25.3.2 (Sep 12, 2017)
- Doom: 2.0.9 (develop 0354edc3a668fdba8b2b1922bcc50cd193dcef5f)
- Graphic display: t (daemon: nil)
- System features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES
- Details:
  ```elisp
  elc count: 0
  uname -a:  Linux wreck 4.13.0-43-generic #48~16.04.1-Ubuntu SMP Thu May 17 12:56:46 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
  modules:   (:feature (evil +everywhere) file-templates (lookup +devdocs +docsets) snippets spellcheck syntax-checker version-control workspaces :completion ivy :ui doom-dashboard doom-modeline doom-quit evil-goggles hl-todo nav-flash (popup +all +defaults) vi-tilde-fringe window-select :tools make password-store pdf rotate-text tmux upload :lang cc crystal clojure csharp data elixir elm emacs-lisp ess go (haskell +intero) hy (java +meghanada) javascript julia latex ledger lua markdown nix ocaml (org +attach +babel +capture +export +present +publish) perl php plantuml purescript python rest ruby rust scala sh swift web :config)
  packages:  n/a
  exec-path: (~/.cargo/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /usr/lib/x86_64-linux-gnu/emacs/25.3/x86_64-linux-gnu ~/.nvm/versions/node/v7.4.0/bin)
  ```
```
</details>

**Comment by hlissner** [comment]
This is certainly a configuration issue on Doom's side. Doom tries very hard to tame smartparens, but there will always be edge cases like these.

Please try the following and let me know if that fixes your issue.

```elisp
(after! smartparens-html
  (setq sp-navigate-consider-sgml-tags nil))
```

**Comment by skosch** [comment]
Yes, that fixes it :)

**Comment by bitwalker** [comment]
@hlissner I'm getting a similar issue with `elixir-mode`, but the error is:

```
sp--skip-match-p: Symbol's function definition is void: sp-elixir-skip-def-p
```

If you'd like, I can open a new issue for it, since it is a bit different.

**Comment by skosch** [comment]
Oh yeah, that elixir-mode error looks awfully familiar to me as well ...

**Comment by hlissner** [comment]
Could you try removing [this line in modules/lang/elixir/config.el (line 12)](https://github.com/hlissner/doom-emacs/blob/develop/modules/lang/elixir/config.el#L12) and see if that fixes that issue?

**Comment by bitwalker** [comment]
@hlissner That does fix it! I hadn't tried that since I figured it was needed, but so far I haven't noticed any issues. If something comes up I'll open a new issue. Thanks!

**Comment by hlissner** [comment]
Alright, both the original and new sp-elixir-skip-def-p issues should be resolved as of cba5270. Let me know if that isn't the case!
